src/app/room/room.controller.js

```diff
@@ -1,41 +1,39 @@
 export class RoomController {
-  constructor ($firebaseAuth, $stateParams, $scope) {
+  constructor ($firebaseAuth, $stateParams, $scope,$log) {
     'ngInject';
-     $scope.getview=null;
-    console.info(`in room controller in room ${$stateParams.roomKey }`);
-     console.log($stateParams.roomKey);
+    $scope.getview=null;
+    $log.info(`in room controller in room ${$stateParams.roomKey }`);
+    $log.log($stateParams.roomKey);
     var keys = [];
     var key;
     //console.log("hui");
     var userRef = new Firebase("https://vyapi.firebaseio.com/rooms/");
     // console.log("outside file");
     userRef.on('value',(snap)=>{
-     //console.log(snap.val());
+      //console.log(snap.val());
       for (key in snap.val()) {
         if (snap.val().hasOwnProperty(key)) {
           keys.push(key);
         }
       }
-    //console.log(keys[0]);
-     var flag=0;
-     var roomId =$stateParams.roomKey;
-     //console.log(roomId);
-     for(key in keys)
-     {
-         if(keys[key]===roomId)
-         {
-          console.log("success");
+      //console.log(keys[0]);
+      var flag=0;
+      var roomId =$stateParams.roomKey;
+      //console.log(roomId);
+      for(key in keys)
+      {
+        if(keys[key]===roomId)
+        {
+          $log.log("success");
           flag=1;
           break;
```

Comment:
I think the break is to avoid unnecessary checks. Like if the first key is a match , then the flag will be 1 and loop will break. If the break does not exist then it will keep checking until the key list is exhausted